src/linter/ui5Types/bestPractices.ts

```diff
@@ -0,0 +1,381 @@
+			rootViewAsyncFlag = AsyncInterfaceStatus.propNotSet;
+			// @ts-expect-error async is part of RootViewDefFlexEnabled and RootViewDef
+			const isRootViewAsync = rootView.async as boolean | undefined;
+			if (typeof isRootViewAsync === "boolean") {
+				rootViewAsyncFlag = isRootViewAsync ? AsyncInterfaceStatus.true : AsyncInterfaceStatus.false;
+			}
+		}
+
+		if (routing) {
+			routingAsyncFlag = AsyncInterfaceStatus.propNotSet;
+			const isRoutingAsync = routing?.config?.async;
+			if (typeof isRoutingAsync === "boolean") {
+				routingAsyncFlag = isRoutingAsync ? AsyncInterfaceStatus.true : AsyncInterfaceStatus.false;
+			}
+		}
+
+		hasManifestDefinition = !!(componentManifest &&
+		ts.isPropertyAssignment(componentManifest) &&
+		componentManifest.initializer.getText() === "\"json\"");
+	}
+
+	return {
+		routingAsyncFlag,
+		rootViewAsyncFlag,
+		hasAsyncInterface,
+		hasManifestDefinition,
+	};
+}
+
+function extractPropsRecursive(node: ts.ObjectLiteralExpression) {
+	const properties = Object.create(null) as propsRecord;
+
+	node.properties?.forEach((prop) => {
+		if (!ts.isPropertyAssignment(prop) || !prop.name) {
+			return;
+		}
+
+		const key = prop.name.getText();
+		if (prop.initializer.kind === ts.SyntaxKind.TrueKeyword) {
+			properties[key] = {value: true, node: prop.initializer};
+		} else if (prop.initializer.kind === ts.SyntaxKind.FalseKeyword) {
+			properties[key] = {value: false, node: prop.initializer};
+		} else if (prop.initializer.kind === ts.SyntaxKind.NullKeyword) {
+			properties[key] = {value: null, node: prop.initializer};
+		} else if (ts.isObjectLiteralExpression(prop.initializer) && prop.initializer.properties) {
+			properties[key] = {value: extractPropsRecursive(prop.initializer), node: prop.initializer};
+		} else if (ts.isArrayLiteralExpression(prop.initializer)) {
+			const resolvedValue = prop.initializer.elements.map((elem) => {
+				if (!ts.isObjectLiteralExpression(elem)) {
+					return;
+				}
+				return extractPropsRecursive(elem);
+			}).filter(($) => $) as propsRecordValueType[];
+
+			properties[key] = {value: resolvedValue, node: prop.initializer};
+		} else if (
+			(ts.isIdentifier(prop.initializer) ||
+			ts.isNumericLiteral(prop.initializer) ||
+			ts.isStringLiteral(prop.initializer)) &&
+
+			prop.initializer.getText()) {
+			properties[key] = {value: prop.initializer.getText(), node: prop.initializer};
+		} else {
+			// throw new Error("Unhandled property assignment");
+		}
+	});
+	return properties;
+}
+
+function reportResults({
+	analysisResult, reporter, classDesc, manifestContent, resourcePath, context,
+}: {
+	analysisResult: AsyncInterfaceFindType;
+	reporter: SourceFileReporter;
+	context: LinterContext;
+	classDesc: ts.ClassDeclaration;
+	manifestContent: string | undefined;
+	resourcePath: string;
+}) {
+	const {hasAsyncInterface, routingAsyncFlag, rootViewAsyncFlag, hasManifestDefinition} = analysisResult;
+
+	if (!hasManifestDefinition) {
+		reporter.addMessage({
```

Comment:
This message is also created when there is no `manifest.json` file next to the `Component.js`. If this is not the case, the suggestion of the message does not apply.

Reply:
Thanks!
In such a case I suppose we need to enforce a fatal error- no definition and no manifest. 
Would UI5 be able to execute the app without a manifest at all?

Reply:
A `Component.js` can define a manifest inline, which should not create a message - I think that's covered already.
In addition, there are fields in the `metadata` (not within `manifest`), like `rootView` that can be used and at least my understanding is that although it is not the best practice, it will continue to work with 2.x. So at least for now I wouldn't expect any message in such a case.

Reply:
Thanks!
Now it's implemented that way.